services/user-tenant-service/src/test/java/com/sourcefuse/jarc/services/usertenantservice/integration/GroupControllerTests.java

```diff
@@ -0,0 +1,504 @@
+@ExtendWith(MockitoExtension.class)
+ class GroupControllerTests {
+
+  @Mock
+  private GroupRepository groupRepository;
+
+  @Mock
+  private UserGroupsRepository userGroupsRepository;
+
+  @InjectMocks
+  private GroupController groupController;
+
+  private Group group;
+  private UserGroup userGroup;
+  private UUID mockGroupId;
+  private MockMvc mockMvc;
+  private String basePath = "/groups";
+
+  @BeforeEach
+  public void setup() {
+    mockMvc = MockMvcBuilders.standaloneSetup(groupController).build();
+    //prepare the test data
+    userGroup = new UserGroup();
+
+    group = MockGroup.getGroupObj();
+    mockGroupId = MockGroup.GROUP_ID;
+  }
+
+  @Test
+  @DisplayName("Create Groups - Success")
+   void testCreateGroups_Success() throws Exception {
+    // Mock the necessary objects and methods
+    Group savedGroup = new Group();
+    savedGroup.setId(MockGroup.GROUP_ID);
+    savedGroup.setName("Test");
+    MockCurrentUserSession.setCurrentLoggedInUser(null, null, null);
+
+    Mockito.when(groupRepository.save(any(Group.class))).thenReturn(savedGroup);
+    Mockito
+      .when(userGroupsRepository.save(any(UserGroup.class)))
+      .thenReturn(userGroup);
+
+    // Call the method under test
+    mockMvc
+      .perform(
+        MockMvcRequestBuilders
+          .post(basePath)
+          .contentType(MediaType.APPLICATION_JSON)
+          .accept(MediaType.APPLICATION_JSON)
+          .content(asJsonString(group))
+      )
+      .andExpect(MockMvcResultMatchers.status().isCreated())
+      .andExpect(
+        MockMvcResultMatchers
+          .jsonPath("$.id")
+          .value(savedGroup.getId().toString())
+      )
+      .andExpect(
+        MockMvcResultMatchers.jsonPath("$.name").value(savedGroup.getName())
+      );
+
+    // Verify the expected interactions and assertions
+    Mockito.verify(groupRepository, Mockito.times(1)).save(any(Group.class));
+    Mockito
+      .verify(userGroupsRepository, Mockito.times(1))
+      .save(any(UserGroup.class));
+  }
+
+  @Test
+  @DisplayName("Test case Should pass for database error")
+  void testCreateGroups_DatabaseError() throws Exception {
+    // Mock a database error when saving the group
+    Mockito
+      .when(groupRepository.save(any(Group.class)))
+      .thenThrow(new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR));
+
+    mockMvc
+      .perform(
+        MockMvcRequestBuilders
+          .post(basePath)
+          .contentType(MediaType.APPLICATION_JSON)
+          .content(asJsonString(group))
+      )
+      .andExpect(MockMvcResultMatchers.status().isInternalServerError())
+      .andExpect(result ->
+        Assertions.assertTrue(
+          result.getResolvedException() instanceof ResponseStatusException
+        )
+      )
+      .andExpect(result ->
+        Assertions.assertEquals(
+          HttpStatus.INTERNAL_SERVER_ERROR.value(),
+          result.getResponse().getStatus()
+        )
+      );
```

Comment:
Not Needed as `      .andExpect(MockMvcResultMatchers.status().isInternalServerError())`
is doing the same thing